### PR TITLE
Refactor and add more unit tests for disktensors

### PR DESCRIPTION
**Pull request by Qazalin**
This refactors existing disktensor movement ops tests and adds new ones for `reshape`

### CODE REVIEW

File: test/unit/test_disk_tensor.py [hunk 1]
```diff
@@ -128,32 +134,20 @@ def test_write_ones(self):
...
-    print(tst)
-    np.testing.assert_allclose(tst, np.arange(10, 20))
-
   def test_assign_slice(self):
-    pathlib.Path(temp("dt4")).unlink(missing_ok=True)
-    cc = Tensor.arange(10, device="CPU").to(f"disk:{temp('dt4')}").realize()
+    def assign_slice(x, slice, val):
+      x[slice] = val
+      return x[slice]
+    helper_test_disk_tensor("dt3", [0,1,2,3], lambda x: assign_slice(x, slice(0,2), [13, 12]), lambda x: x[0:2].assign([13, 12]))
```
Comment (geohot) [inline test/unit/test_disk_tensor.py:141]: You shouldn't need a different numpy function here (or anywhere)

Comment (Qazalin) [reply test/unit/test_disk_tensor.py]: yeah some like reshape dont need a new one - though I couldn't find an api like `.assign` or `.permute`

File: test/unit/test_disk_tensor.py [hunk 2]
```diff
@@ -128,32 +135,17 @@ def test_write_ones(self):
...
-    slice_me = Tensor.empty(10, 10, device=f"disk:{temp('dt5')}")
-    tst = slice_me[1].numpy()
-    print(tst)
-    np.testing.assert_allclose(tst, np.arange(10, 20))
-
   def test_assign_slice(self):
-    pathlib.Path(temp("dt4")).unlink(missing_ok=True)
-    cc = Tensor.arange(10, device="CPU").to(f"disk:{temp('dt4')}").realize()
+    helper_test_disk_tensor("dt3", [0,1,2,3], lambda x: (x.__setitem__(slice(0,2), [13, 12]), x[0:2])[1], lambda x: x[0:2].assign([13, 12]))
+    helper_test_disk_tensor("dt4", [[0,1,2,3],[4,5,6,7]], lambda x: ((x.__setitem__(slice(0,1), [[13, 12, 11, 10]]), x[0:1])[1]), lambda x: x[0:1].assign([[13, 12, 11, 10]]))
```
Comment (geohot) [inline test/unit/test_disk_tensor.py:140]: Not what I mean. Why doesn't `x[0:2].assign([13,12])` work on disktensors

File: test/unit/test_disk_tensor.py [hunk 3]
```diff
@@ -128,32 +135,17 @@ def test_write_ones(self):
...
+  def test_reshape(self):
+    helper_test_disk_tensor("dt5", [1,2,3,4,5], lambda x: x.reshape((1,5)))
+    helper_test_disk_tensor("dt6", [1,2,3,4], lambda x: x.reshape((2,2)))
 
-    #cc.assign(np.ones(10)).realize()
-    print(cc[3:5].numpy())
-    cc[3:5].assign([13, 12]).realize()
-    print(cc.numpy())
+  def test_permute(self):
+    helper_test_disk_tensor("dt7", [[1,2,3,4,5],[6,7,8,9,10]], lambda x: np.transpose(x, (0,1)), lambda x: x.permute(0,1))
```
Comment (geohot) [inline test/unit/test_disk_tensor.py:147]: There's no way this works.

File: test/unit/test_disk_tensor.py (hunk 2 above)
Comment (Qazalin) [reply test/unit/test_disk_tensor.py]: needed a tiny `__setitem__` implementation in `tensor.py`

File: test/unit/test_disk_tensor.py (hunk 3 above)
Comment (Qazalin) [reply test/unit/test_disk_tensor.py:150]: yeah.. `as_strided` just worked in this case

Comment (geohot) [reply test/unit/test_disk_tensor.py]: Oh...because that's a fake NOOP permute...

**Pull request merged**